tests/integration/test_kalman_filter_with_sparse_sites.py

```diff
@@ -0,0 +1,56 @@
+import numpy as np
+import tensorflow as tf
+import pytest
+from gpflow.config import default_float
+
+from markovflow.kernels.matern import Matern12
+from markovflow.mean_function import LinearMeanFunction
+from markovflow.models.gaussian_process_regression import GaussianProcessRegression
+from markovflow.kalman_filter import KalmanFilterWithSparseSites, UnivariateGaussianSitesNat
+from markovflow.likelihoods import MultivariateGaussian
+
+
+@pytest.fixture(name="kalman_gpr_setup")
+def _setup(batch_shape):
+    """
+    FIXME: Currently batch_shape isn't used.
+    """
+    time_grid = np.arange(0.0, 1.0, 0.01)
```

Comment:
can you do an inhomogenous time grid?
Do tests still pass?
```suggestion
    dt = 0.01
    time_grid = np.arange(0.0, 1.0, dt)
    time_grid += np.random.rand(*time_grid.shape) * dt / 2. 
```

Reply:
It works! Added test cases for that too.

Reply:
I just coded it up a bit differently.